packages/angular_devkit/core/src/json/schema/registry.ts

```diff
@@ -175,39 +195,138 @@ export class CoreSchemaRegistry implements SchemaRegistry {
       return {};
     }
 
-    // tslint:disable-next-line:no-any
-    const id = (validate.schema as any).$id || (validate.schema as any).id;
-    let fullReference = (ref[0] === '#' && id) ? id + ref : ref;
+    let refMap = validate as AjvRefMap;
+    const rootRefMap = validate.root as AjvRefMap;
+
+    // Resolve from the root if it's different.
+    if (validate.root && validate.schema !== rootRefMap.schema) {
```

Comment:
Why always use the root?  Would that break relative refs within subschemas?

Reply:
Doesn't look like it. We only use the root if it's available. The root AFAIK will _always_ have the new refs as they're resolved (in async) before calling this method, and added to the root.

Reply:
Works for me then.